src/sentry/eventstore/models.py

```diff
@@ -157,45 +180,354 @@ def get_minimal_user(self):
                 {"id": user_id, "email": email, "username": username, "ip_address": ip_address}
             )
 
-        return super(Event, self).get_minimal_user()
+        return self.get_interface("user")
 
-    # If the data for these is available from snuba, we assume
-    # it was already normalized on the way in and we can just return
-    # it, otherwise we defer to EventCommon implementation.
     def get_event_type(self):
+        """
+        Return the type of this event.
+
+        See ``sentry.eventtypes``.
+        """
         column = self.__get_column_name(Columns.TYPE)
         if column in self._snuba_data:
             return self._snuba_data[column]
-        return super(Event, self).get_event_type()
+        return self.data.get("type", "default")
 
     @property
     def ip_address(self):
         column = self.__get_column_name(Columns.USER_IP_ADDRESS)
         if column in self._snuba_data:
             return self._snuba_data[column]
-        return super(Event, self).ip_address
+
+        ip_address = get_path(self.data, "user", "ip_address")
+        if ip_address:
+            return ip_address
+
+        remote_addr = get_path(self.data, "request", "env", "REMOTE_ADDR")
+        if remote_addr:
+            return remote_addr
+
+        return None
 
     @property
     def title(self):
         column = self.__get_column_name(Columns.TITLE)
         if column in self._snuba_data:
             return self._snuba_data[column]
-        return super(Event, self).title
+
+        et = eventtypes.get(self.get_event_type())()
+        return et.get_title(self.get_event_metadata())
 
     @property
     def culprit(self):
         column = self.__get_column_name(Columns.CULPRIT)
         if column in self._snuba_data:
             return self._snuba_data[column]
-        return super(Event, self).culprit
+        return self.data.get("culprit")
 
     @property
     def location(self):
         column = self.__get_column_name(Columns.LOCATION)
         if column in self._snuba_data:
             return self._snuba_data[column]
-        return super(Event, self).location
+        et = eventtypes.get(self.get_event_type())()
+        return et.get_location(self.get_event_metadata())
+
+    @classmethod
+    def generate_node_id(cls, project_id, event_id):
+        """
+        Returns a deterministic node_id for this event based on the project_id
+        and event_id which together are globally unique. The event body should
+        be saved under this key in nodestore so it can be retrieved using the
+        same generated id when we only have project_id and event_id.
+        """
+        return md5("{}:{}".format(project_id, event_id)).hexdigest()
+
+    # TODO (alex) We need a better way to cache these properties.  functools32
```

Comment:
Yes, let's get alex to fix it. (feel free to remove his name from the todo)

Reply:
Removed